app/src/components/domain/RiskImminentEvents/WfpAdam/index.tsx

```diff
@@ -83,6 +117,46 @@ function WfpAdam(props: Props) {
         apiType: 'risk',
         url: '/api/v1/adam-exposure/{id}/exposure/',
         pathVariables: ({ eventId }) => ({ id: Number(eventId) }),
+        onSuccess: (res) => {
+            if (isNotDefined(res)) {
+                return defaultLayersValue;
+            }
+
+            const { storm_position_geojson } = res;
+
+            if (isNotDefined(storm_position_geojson)) {
+                return defaultLayersValue;
+            }
+
+            const stormPositions = isValidFeatureCollection(storm_position_geojson)
+                ? storm_position_geojson
+                : undefined;
+
+            const updatedLayers = {} as typeof defaultLayersValue;
+
+            stormPositions?.features?.reduce((_, feature) => {
+                if (feature.geometry.type === 'Point' || feature.geometry.type === 'MultiPoint') {
+                    updatedLayers[LAYER_CYCLONE_NODES] = true;
+                }
+
+                if (feature.geometry.type === 'LineString' || feature.geometry.type === 'MultiLineString') {
+                    updatedLayers[LAYER_CYCLONE_TRACKS] = true;
+                }
+                if (feature.geometry.type === 'Polygon' || feature.geometry.type === 'MultiPolygon') {
+                    updatedLayers[LAYER_CYCLONE_BUFFERS] = true;
+                }
+
+                if (feature.properties?.alert_level === 'Cones') {
+                    updatedLayers[LAYER_CYCLONE_UNCERTAINTY] = true;
+                }
+                return updatedLayers as typeof defaultLayersValue;
```

Comment:
We could extract this logic into a function and reuse for other sources as well. Also, we don't need to use reduce here.